### PR TITLE
Add Treatment Report Functionality

### LINKED ISSUE #31: Add Functionality for the Treatment Record
The second part of the treatment database is called the treatment record and requires the following fields.  Unless otherwise noted, these are text fields.

## Description

- General Remarks

- Binding

- Textblock

- Primary Support

- Medium

- Attachments | Inserts

- Housing

## Condition

- Summary

- Binding 

- Textblock

- Primary Support

- Medium

- Housing (Controlled Vocabulary)

- Housing Naritive

- Attachments | Inserts

- Previous Treatment

- Materials Analysis

## Treatment Proposal

- Proposal

- Housing Need (Controlled Vocabulary)

- Factors Influencing Treatment

- Performed Treatment

- Housing Provided (Controlled Vocabulary)

- Housing Narrative

- Storage and Handling Notes

- Total Treatment Time (integer)

### PR DESCRIPTION
**Pull request by chuckgreenman**
closes #31 ;

Adds the 1 to 1 relation between the conservation report and the treatment report, set up forms.  I still need to write some tests for this though.

### CODE REVIEW

**Comment by chuckgreenman** [comment]
@hortongn I think this is ready to rock and roll now!

**Comment by chuckgreenman** [comment]
Spoke too soon, something is broken.

**Comment by chuckgreenman** [comment]
@hortongn While working to resolve rubocop issues I inverted a boolean expressions value unintentionally, causing unexpected behavior, it is now fixed and ready for review.

File: app/views/conservation_records/show.html.erb [hunk 1]
```diff
@@ -122,8 +122,165 @@
         <% end %>
       </ul>
     <% end %>
-    
   </div>
+
+  <br>
```
Comment (crowesn) [inline app/views/conservation_records/show.html.erb]: Probably want to hide form from read_only users.

File: app/controllers/treatment_reports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
```
Comment (crowesn) [inline app/controllers/treatment_reports_controller.rb]: I think it would be useful to throw up a :success flash message on create and update.

**Comment by chuckgreenman** [comment]
@crowesn Thanks for the review!
Updated to catch up with QA.  Hid the treatment report from non-write users and added flash messages.

**Comment by crowesn** [comment]
I'll go ahead and merge this but I'm thinking now we should add more nuance to how we display the treatment report form; like maybe grey it out for non-permitted users. I'll open an issue. 

Nice work @chuckgreenman !

**Pull request merged**